src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -23,7 +24,7 @@ object SymbolStoreEntry {
 case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) {
 
   wdlValue match {
-    case Some (value) => wdlType.checkCompatible(value.value)
+    case Some (value) => if(wdlType != value.wdlType) throw new UnsupportedOperationException("Fix this.")
```

Comment:
I think we can definitely implement checks to ensure that this won't actually happen.  If we didn't do the check here, we'd have to check that the types that the user provided are correct and we'd have to check that the types on all `TaskOutputs` are valid when we process those.